### DISCUSSION TITLE
Will a forked repo be accepted?

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#25476] [General]
The lib (cppjieba) is not popular, and it depneds on an even less popular lib (limonp) and gtest.
In the original lib, the depnedencies are simply copied there, including limonp and gtest.

It needs many refactors, so I want to fork it and refactor it, will vcpkg accept a forked repo like this?

**Comment by playgithub** [comment]
OK, it will, better be marked as unofficial.

**Comment by ghost** [comment]
@playgithub  did you get any answer regarding this offline ? i am also thinking forking other opensource projects.

**Reply by dg0yt** [reply]
IMO forks are not acceptable. It is not feasible to permanently review how they diverge from upstream. 
But forks can be a way to maintain patches. (Actually it is not much different from what I do ad-hoc locally when updating ports.)

**Reply by playgithub** [reply]
However, port `poppler` is official initially, but it becomes unofficial now, e.g.
```cmake
find_package(unofficial-poppler REQUIRED)
target_link_libraries(${PROJECT_NAME}
    PRIVATE
    unofficial::poppler::poppler-cpp
```

**Reply by dg0yt** [reply]
This is not an unofficial forked repo. It is unofficial CMake config. And the patch is right here in vcpkg.

**Reply by playgithub** [reply]
Is it better to be compatible with the official cmake interface as below
```cmake
find_package(Poppler REQUIRED)
target_link_libraries(${PROJECT_NAME}
    PRIVATE
    ${Poppler_LIBRARIES}
```

**Reply by dg0yt** [reply]
There is no such official interface for Poppler. That's the reason why vcpkg provides a private/unofficial config. The `unofficial` prefix is a warning that these identifiers will be dropped when Poppler starts to export official config.

**Reply by dg0yt** [reply]
Read more:
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#add-cmake-exports-in-an-unofficial--namespace

**Reply by playgithub** [reply]
> There is no such official interface for Poppler. That's the reason why vcpkg provides a private/unofficial config. The `unofficial` prefix is a warning that these identifiers will be dropped when Poppler starts to export official config.

OK, it's in extra cmake modules by KDE (https://api.kde.org/ecm)

**Reply by dg0yt** [reply]
> OK, it's in extra cmake modules by KDE (https://api.kde.org/ecm)

This is a third-party Find module. And it shares the disadvantages of find modules for non-trivial packages:
- useless for static linkage because unable to know the transitive usage requirements (which depend on configuration), unless it can get them from pkg-config
- unaware of multi-config configuration (release+debug).

But let's stop here. The CMake package topic is unrelated to the forked repo topic.

**Reply by Neumann-A** [reply]
There is already a port for ecm, if i remeber correctly. It is just called differently
